src/settings/SettingsStore.js

```diff
@@ -220,34 +226,54 @@ export default class SettingsStore {
 
         const handlers = SettingsStore._getHandlers(settingName);
 
+        // Check if we need to invert the setting at all. Do this after we get the setting
+        // handlers though, otherwise we'll fail to read the value.
+        let inverted = false;
+        if (setting.invertedSettingName) {
+            //console.warn(`Inverting ${settingName} to be ${setting.invertedSettingName} - legacy setting`);
+            settingName = setting.invertedSettingName;
+            inverted = true;
+        }
+
         if (explicit) {
             const handler = handlers[level];
-            if (!handler) return SettingsStore._tryControllerOverride(settingName, level, roomId, null, null);
-            const value = handler.getValue(settingName, roomId);
-            return SettingsStore._tryControllerOverride(settingName, level, roomId, value, level);
+            if (!handler) {
+                let value = SettingsStore._tryControllerOverride(setting, level, roomId, null, null);
+                if (inverted) value = !value;
```

Comment:
Hmm, slightly concerned about this important line being duplicated 4x in the code below...